src/unknown_0255E0.c

```diff
@@ -886,7 +886,106 @@ s32 func_8002B9BC(Object *obj, f32 *arg1, f32 *arg2, s32 arg3) {
 }
 
 GLOBAL_ASM("asm/non_matchings/unknown_0255E0/func_8002BAB0.s")
+
+#ifdef NON_EQUIVALENT
+
+// Maximum size for a level model is 522.5 KiB
+#define LEVEL_MODEL_MAX_SIZE 0x82A00
+
+#define LOCAL_OFFSET_TO_RAM_ADDRESS(ptr) \
+    ptr = (s32)((u8*)ptr) + (s32)((u8*)gCurrentLevelModel)
+
+// Loads a level track from the index in the models table.
+// Has regalloc issues.
+void func_8002C0C4(s32 modelId) {
+    s32 i, j, k;
+    s32 numLevelModels;
+    s32 temp_s4;
+    s32 offset, size;
+    u8 *compressedRamAddr;
+    LevelModelSegment *segment;
+    TriangleBatchInfo *batch;
+    Vertex *vertex;
+    
+    func_8007B374(0xFF00FF);
+    D_8011D30C = allocate_from_main_pool_safe(LEVEL_MODEL_MAX_SIZE, 0xFFFF00FFU);
+    gCurrentLevelModel = D_8011D30C;
+    D_8011D370 = allocate_from_main_pool_safe(0x7D0, 0xFFFF00FFU);
+    D_8011D374 = allocate_from_main_pool_safe(0x1F4, 0xFFFF00FFU);
+    D_8011D378 = 0;
+    D_8011D310 = load_asset_section_from_rom(ASSET_LEVEL_MODELS_TABLE);
+    for(numLevelModels = 0; D_8011D310[numLevelModels] != -1; numLevelModels++) {}
+    numLevelModels--;
+    if (modelId >= numLevelModels) {
+        modelId = 0;
+    }
+    offset = D_8011D310[modelId];
+    size = D_8011D310[modelId + 1] - offset;
+    compressedRamAddr = (u8*)gCurrentLevelModel + (LEVEL_MODEL_MAX_SIZE - size);
+    compressedRamAddr = compressedRamAddr - (u8*)((s32)compressedRamAddr % 16);
```

Comment:
```suggestion
    compressedRamAddr = (u8*)gCurrentLevelModel + LEVEL_MODEL_MAX_SIZE - size;
    compressedRamAddr = compressedRamAddr - (u8*)((s32)compressedRamAddr % 16);
```
Second line is also weird. Does it still ~match to do this?
```
    compressedRamAddr -= (s32)compressedRamAddr % 16;
```